client/input.go

```diff
@@ -14,10 +14,26 @@ import (
 // MinPassLength is the minimum acceptable password length
 const MinPassLength = 8
 
+var currentStdin *bufio.Reader
+
+func init() {
+	currentStdin = bufio.NewReader(os.Stdin)
+}
+
 // BufferStdin is used to allow reading prompts for stdin
 // multiple times, when we read from non-tty
 func BufferStdin() *bufio.Reader {
-	return bufio.NewReader(os.Stdin)
+	return currentStdin
+}
+
+// OverrideStdin allows to temporarily override stdin
+func OverrideStdin(newStdin *bufio.Reader) (cleanUp func()) {
```

Comment:
why do we need to do this? Why the need for a global?

Reply:
This is an intermediate solution to replace stdin (which was a singleton anyway).
The idea is to move all bufio to cli context so I/O is completely decoupled.
This change will come in a next PR (#3558)